content/testing.md

```diff
@@ -0,0 +1,23 @@
+# Testing Guide
+
+TODO: Explain why it is essential to have proper testing in our projects and for PRs to maintain tests up-to-date.
+
+## Unit Tests
+
+TODO: Introduce the subject with the example of how we made the Flow DPS mapper testable.
+TODO: Explain internal/external unit tests.
+TODO: Explain manual mocks.
+TODO: Explain pseudorandom generic values.
+TODO: Explain test parallelization.
+TODO: Explain how awesome the native `testing` package is.
+TODO: Explain subtest and structure of nested tests.
+TODO: Explain when to use subtests vs when to use table-driven tests.
+
+## Integration Tests
+
+TODO: Explain build tag usage to prevent integration tests from being run at all times.
```

Comment:
https://medium.com/@tharun208/build-tags-in-go-f21ccf44a1b8
https://mickey.dev/posts/go-build-tags-testing/
https://dave.cheney.net/2019/10/06/use-internal-packages-to-reduce-your-public-api-surface

I.E. Talk about the particular use of build tags for our case and mention one or more of the articles if they want to learn more about it.